### DISCUSSION TITLE
Azure site to site proposals

### DISCUSSION THREAD

**Discussion by trx-jay** [strongswan/strongswan#1275] [General]
Hi Folks,

Noob here trying to finish a project here, I think I have it bar these damn proposals, this is for an Linux Ubuntu VPC Machine to Azure Cloud network interface, only thing I can't suss out is how to config my IKE and ESP at my end to match the Below

IKEv2 Encryption | GCMAES256, AES256
IKEv2 Integrity | GCMAES256, SHA384, SHA256, SHA1
DH Group | DHGroup2
IPsec Encryption | GCMAES256, AES256
IPsec Integrity | GCMAES256, SHA256
PFS Group | None, PFS24, ECP384, ECP256, PFS2048, PFS2, PFS1

Every vairation I've tried so far
` #ike=chacha20poly1305-sha512-curve25519-prfsha512,aes256gcm16-sha384-prfsha384-ecp384,aes256-sha1-modp1024,aes128-sha1-modp1024,3des-sha1-modp1024!
   #ike=3des-sha1-modp1024!
   ike=aes256-sha256-modp1024,aes256-sha1-modp1024!
   #esp=chacha20poly1305-sha512,aes256gcm16-ecp384,aes256-sha256,aes256-sha1,3des-sha1!
   #esp=aes256-sha256-modp1024!
   esp=aes256-sha256-modp1024!
`

Any help would be so helpful!

**Comment by Thermi** [comment]
ike=aes256gcm16-prfsha256-modp1024

esp=aes256gcm16-ecp256

**Reply by trx-jay** [reply]
Hi Thermi,

Thanks for the reply, my returned output gave me -

- charon[202192]: 09[CFG] algorithm 'prfsha26' not recognized
- charon[202192]: 09[CFG] skipped invalid proposal string: aes256gcm16-prfsha26-modp1024

I've checked the docs and I can't find that string anywhere? Any suggestions?

**Comment by Thermi** [comment]
Hi, that was a typo in the original comment that I fixed seconds after sending it.

Am 6. September 2022 08:43:04 UTC schrieb trx-jay ***@***.***>:
>Hi Thermi,
>
>Thanks for the reply, my returned output gave me -
>
>- charon[202192]: 09[CFG] algorithm 'prfsha26' not recognized
>- charon[202192]: 09[CFG] skipped invalid proposal string: aes256gcm16-prfsha26-modp1024
>
>I've checked the docs and I can't find that string anywhere? Any suggestions?
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/1275#discussioncomment-3567270
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
-- 
Sent from mobile. Please excuse any typos or brevity

**Reply by trx-jay** [reply]
Ah, thanks very much, I'm waiting on word from the other side I keep getting hit with - 
parsed IKE_AUTH response 1 [ N(MS_STATUS(87)) ]

Which I'd imagine is Azure's response message that they don't match but I think the other side has changed theirs again.

What proposals does the following match to or is that a silly question?
 ike=aes256gcm16-prfsha256-modp1024
esp=aes256gcm16-ecp256

My biggest issue I think has been with structuring these, do they follow a packet structure?  Or can they be kind of random in their positioning?

The other side has came back with their latest changes which I only found out they changed again -
`
"ipsecEncryption": "AES256",

                "ipsecIntegrity": "SHA256",

                "ikeEncryption": "AES256",

                "ikeIntegrity": "SHA256",

                "dhGroup": "DHGroup14",

                "pfsGroup": "None"
`
So I've set mine simply to - 
 ike=aes256-sha256-modp2048
 esp=aes256-sha256

But I keep getting parsed IKE_AUTH response 1 [ N(MS_STATUS(87)) ] as a response in my logs and 
Sep  6 16:11:01 NetZero charon: 14[MGR] checkout IKEv2 SA by message with SPIs c17aff29e295a4d2_i 11bb5f0df4c40961_r
Sep  6 16:11:01 NetZero charon: 14[MGR] IKE_SA checkout not successful

**Reply by trx-jay** [reply]
I should state this is a PSK that both parties have and are allowing the sending of packets but the connection never goes up or properly connects

**Reply by Thermi** [reply]
The proposal I provided conforms to 
```
IKEv2 Encryption | GCMAES256
IKEv2 Integrity | GCMAES256
DH Group | DHGroup2
IPsec Encryption | GCMAES256
IPsec Integrity | GCMAES256
PFS Group | ECP256
```


> My biggest issue I think has been with structuring these, do they follow a packet structure? Or can they be kind of random in their positioning?

They need to be together.
```
But I keep getting parsed IKE_AUTH response 1 [ N(MS_STATUS(87)) ] as a response in my logs and
Sep 6 16:11:01 NetZero charon: 14[MGR] checkout IKEv2 SA by message with SPIs c17aff29e295a4d2_i 11bb5f0df4c40961_r
Sep 6 16:11:01 NetZero charon: 14[MGR] IKE_SA checkout not successful
```

Please provide complete logs as described in #196

**Reply by trx-jay** [reply]
I'm patiently waiting on word of potential split tunnelling and forwarding issues

**Reply by Thermi** [reply]
MS_STATUS(87) is WIN32 error code 0x57 (ERROR_INVALID_PARAMETER), so I suppose something is wrong
